analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -461,10 +463,19 @@ class Pub(
                     var vcs = VcsInfo.EMPTY
                     var authors: SortedSet<String> = sortedSetOf<String>()
 
-                    // For now, we ignore SDKs like the Dart SDK and the Flutter SDK in the analyzer.
                     when {
+                        pkgInfoFromLockFile["source"].textValueOrEmpty() == "path" -> {
+                            rawName = packageName
+                            val path = pkgInfoFromLockFile["description"]["path"].textValueOrEmpty()
+                            vcs = VersionControlSystem.forDirectory(workingDir.resolve(path))?.getInfo() ?: run {
+                                log.warn { "Invalid path of package $rawName: $path is outside of the project root $workingDir" }
```

Comment:
Nits: Please single-quite `'$path'` and `'$workingDir'` in case they contain spaces, and end the log sentence with a dot.

Reply:
And @bennati please also look into the *detekt* issues about too long lines.

Reply:
done

Reply:
There now seems to be [some breakage](https://github.com/oss-review-toolkit/ort/runs/6421480451?check_suite_focus=true#step:4:256).

Reply:
The issue with `workingDir` not being passed is related to this https://github.com/oss-review-toolkit/ort/pull/5330#discussion_r868175788
How can I mark it as an optional parameter?

Reply:
Kept the parameter as is and added it to the calls in the other tests